gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -252,6 +248,43 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     end
   end
 
+  def self._handle_missing_method_signature(mod, new_method, method_name, obj, *args, &blk)
+    original_name = new_method.original_name
+    method_sig = T::Private::Methods.signature_for_method(mod.instance_method(original_name)) if original_name
+
+    if method_sig
+      # We're handling a case where `alias_method` was called for a method
+      # which had already had a `sig` applied.
+      #
+      # Note, this logic is duplicated above, make sure to keep changes in sync.
+      if method_sig.check_level == :always || (method_sig.check_level == :tests && T::Private::RuntimeLevels.check_tests?)
+        # Checked, so copy the original signature to the aliased copy.
+        T::Private::Methods.unwrap_method(mod, method_sig, new_method)
+      else
+        # Unchecked, so just make `alias_method` behave as if it had been called pre-sig.
+        mod.send(:alias_method, method_name, original_name)
+      end
+    else
+      msg = "`sig` not present for method `#{method_name}` but you're trying to run it anyways. " \
+        "This should only be executed if you used `alias_method` to grab a handle to a method after `sig`ing it, but that clearly isn't what you are doing. " \
+        "Maybe look to see if an exception was thrown in your `sig` lambda or somehow else your `sig` wasn't actually applied to the method. " \
+        "Contact #dev-productivity if you're really stuck."
```

Comment:
Is this messages still relevant?

Reply:
Not really, I think!

Also I removed the soft-assert fallback because it was broken (I'd stopped threading through the return value) and without any way to cover that with tests it didn't seem worth maintaining.